Sources/LanguageServerProtocol/Types/LineIterator.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  LineIterator.swift
+//  LanguageServerProtocol
+//
+//  Created by Ryan Lovelett on 5/30/18.
+//
+
+private extension Character {
+
+    var isLineEnding: Bool {
+        return self == "\r\n" || self == "\r" || self == "\n"
+    }
+
+}
+
+struct Line {
+    let number: Int
+    let start: String.Index
+    let end: String.Index
+    let last: Bool
+
+    func contains(_ index: String.Index) -> Bool {
+        if last {
+            return (start...end).contains(index)
+        } else {
+            return (start..<end).contains(index)
+        }
+    }
+}
+
+struct LineIterator: IteratorProtocol {
+
+    private let text: String
+
+    private var cursor: String.Index
+
+    private var lineNumber: Int
+
+    private var handleTrailingNewLine: Bool
```

Comment:
Document what `handleTrailingNewLine` is really trying to do. Probably needs a better name too.